src/storage/mvcc/txn.rs

```diff
@@ -253,13 +272,22 @@ impl<S: Snapshot, P: PdClient + 'static> MvccTxn<S, P> {
                 "async commit is not yet compatible with large transactions"
             );
 
-            // TODO(nrc) this is going to block all the other keys' processing and writing, we should
-            // do it async.
-            // TODO(nrc) this is also unsound! If we don't complete taking the lock until after another
-            // node gets a start ts to read the key, it can violate the snapshot property.
-            let ts = ::futures_executor::block_on(Compat01As03::new(self.pd_client.get_tso()))?;
-            lock.min_commit_ts = ts;
+            // This operation should not block because the latch makes sure only one thread
+            // is operating on this key.
+            let key_guard =
+                CONCURRENCY_MANAGER_LOCK_DURATION_HISTOGRAM.observe_closure_duration(|| {
```

Comment:
Getting an instant takes about 20ns. Adding it to histogram is also an sub-microsecond operation. So this won't increase much latency.
This metric is to see the impact of the BIG Mutex out of the BTreeMap. It can be removed when we improve the concurrency of the lock table.